### PR TITLE
refactor: build wmstore and wmstorewatcher directly, and remove some unnecessary fields 

### PR DESCRIPTION
**Pull request by whynowy**
1. Functions to build `WatermarkStore` and `WatermarkStoreWatcher` directly from `bucket`.
2. Remove `bucket` (only for logging) as an arg, use `ctx` to pass it.
3. Remove `pipeline` (only for logging) as an arg, use `ctx` to pass it.

### CODE REVIEW

File: pkg/shared/kvs/inmem/kv_watch.go [hunk 1]
```diff
@@ -42,15 +41,14 @@ type inMemWatch struct {
...
+func NewInMemWatch(ctx context.Context, bucketName string, kvEntryCh <-chan kvs.KVEntry) (kvs.KVWatcher, error) {
 	k := &inMemWatch{
-		pipelineName: pipelineName,
 		bucketName:   bucketName,
 		kvEntryCh:    kvEntryCh,
 		kvHistory:    []kvs.KVEntry{},
 		updatesChMap: make(map[string]chan kvs.KVEntry),
 		doneCh:       make(chan struct{}),
-		log:          logging.FromContext(ctx).With("pipeline", pipelineName).With("bucketName", bucketName),
+		log:          logging.FromContext(ctx).With("bucketName", bucketName),
```
Comment (whynowy) [inline pkg/shared/kvs/inmem/kv_watch.go:51]: Only for logging, removed.

File: pkg/shared/kvs/jetstream/kv_store.go [hunk 2]
```diff
@@ -42,11 +42,11 @@ type jetStreamStore struct {
 var _ kvs.KVStorer = (*jetStreamStore)(nil)
 
 // NewKVJetStreamKVStore returns KVJetStreamStore.
-func NewKVJetStreamKVStore(ctx context.Context, pipelineName string, kvName string, client *jsclient.NATSClient, opts ...JSKVStoreOption) (kvs.KVStorer, error) {
+func NewKVJetStreamKVStore(ctx context.Context, kvName string, client *jsclient.NATSClient, opts ...JSKVStoreOption) (kvs.KVStorer, error) {
 	var err error
 	var jsStore = &jetStreamStore{
 		client: client,
-		log:    logging.FromContext(ctx).With("pipeline", pipelineName).With("kvName", kvName),
+		log:    logging.FromContext(ctx).With("kvName", kvName),
```
Comment (whynowy) [inline pkg/shared/kvs/jetstream/kv_store.go:49]: The log in `ctx` comes with `pipeline`.

File: pkg/shared/kvs/jetstream/kv_watch.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func NewKVJetStreamKVWatch(ctx context.Context, pipelineName string, kvName stri
 		kvwTimer: time.NewTimer(kvOpts.watcherCreationThreshold),
 		opts:     kvOpts,
 		doneCh:   make(chan struct{}),
-		log:      logging.FromContext(ctx).With("pipeline", pipelineName).With("kvName", kvName),
+		log:      logging.FromContext(ctx).With("kvName", kvName),
```
Comment (whynowy) [inline pkg/shared/kvs/jetstream/kv_watch.go:69]: The log in `ctx` comes with `pipeline`.

File: pkg/watermark/processor/processor_manager.go [hunk 4]
```diff
@@ -81,9 +78,8 @@ func NewProcessorManager(ctx context.Context, watermarkStoreWatcher store.Waterm
 		otWatcher:                watermarkStoreWatcher.OffsetTimelineWatcher(),
 		heartbeat:                NewProcessorHeartbeat(),
 		processors:               make(map[string]*ProcessorToFetch),
-		bucket:                   bucket,
 		fromBufferPartitionCount: fromBufferPartitionCount,
-		log:                      logging.FromContext(ctx).With("bucket", bucket),
+		log:                      logging.FromContext(ctx),
```
Comment (whynowy) [inline pkg/watermark/processor/processor_manager.go:82]: The log from the `ctx` comes with `bucket`.

File: pkg/watermark/timeline/offset_timeline.go [hunk 5]
```diff
@@ -42,13 +42,13 @@ type OffsetTimeline struct {
...
 // NewOffsetTimeline returns OffsetTimeline.
-func NewOffsetTimeline(ctx context.Context, c int, bucket string) *OffsetTimeline {
+func NewOffsetTimeline(ctx context.Context, c int) *OffsetTimeline {
 	// Initialize a new empty watermarks DLL with nil values of the size capacity.
 	// This is to avoid length check: when a new element is added, the tail element will be deleted.
 	offsetTimeline := OffsetTimeline{
 		ctx:      ctx,
 		capacity: c,
-		log:      logging.FromContext(ctx).With("bucket", bucket),
+		log:      logging.FromContext(ctx),
```
Comment (whynowy) [inline pkg/watermark/timeline/offset_timeline.go:51]: The log already has `bucket`.

**Review by vigith — APPROVED** [review]

**Review by yhl25 — APPROVED** [review]

**Pull request merged**